packages/nuxt/src/core/runtime/nitro/utils/error.ts

```diff
@@ -0,0 +1,27 @@
+import { type H3Event, getRequestHeader } from 'h3'
+
+/**
+ * Nitro internal functions extracted from https://github.com/nitrojs/nitro/blob/main/src/runtime/internal/utils.ts
+ */
+
+export function isJsonRequest (event: H3Event) {
+  // If the client specifically requests HTML, then avoid classifying as JSON.
+  if (hasReqHeader(event, 'accept', 'text/html')) {
+    return false
+  }
+  return (
+    hasReqHeader(event, 'accept', 'application/json') ||
+    hasReqHeader(event, 'user-agent', 'curl/') ||
+    hasReqHeader(event, 'user-agent', 'httpie/') ||
+    hasReqHeader(event, 'sec-fetch-mode', 'cors') ||
+    event.path.startsWith('/api/') ||
```

Comment:
FWIW it is a negative and confusing DX (i made a change in nitro based on lots of feedbacks heard from Nuxt users). At least for development we could make new behavior.

Reply:
I'm planning to update for nuxt v4